engines/suppliers/app/controllers/suppliers/api/v1/providers_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Suppliers
+  module Api
+    module V1
+      class ProvidersController < ApplicationController
+        before_action :authenticate
+        before_action :load_group, only: [:index]
+        before_action :load_producer, only: [:show]
+
+        # GET /api/v1/providers?group_id=
+        #
+        # You must pass `group_id` param as filter
+        #
+        def index
+          producer_ids = ::Suppliers::Supplier.where(group_id: @group.id).pluck(:producer_id)
+          providers = ::Suppliers::Producer.where(id: producer_ids)
+
+          render json: ::Suppliers::ProvidersSerializer.new(providers)
+        end
+
+        # GET /api/v1/providers/:id
+        #
+        def show
+          render json: ::Suppliers::ProviderSerializer.new(@producer)
+        end
+
+        private
+
+        def load_group
+          @group = ::Suppliers::Group.find_by_id(params[:group_id])
+
+          return head :not_found unless @group
+
+          authorize @group
+        end
+
+        def load_producer
+          @producer = ::Suppliers::Producer.find_by_id(params[:id])
+
+          return head :not_found unless @producer
+
+          authorize @producer
```

Comment:
Furthermore, since we're in the `Suppliers::ProvidersController`, all the operations that we'll do with `@producer` will be scoped under the `Suppliers` namespace since we're in that engine. In this way if the concept of `Producer` needs some specific business logic, only related to this `Suppliers` context we'll be able to keep it clean qithout polluting the `Producers::Producer` model.

Even the `current_user` here is an instance of `Suppliers::User` and not of `Account::User` (which is the actual *owner* of the `users` table).

Better? Useful?